src/test/java/com/google/neighborgood/IntegrationTest.java

```diff
@@ -0,0 +1,1065 @@
+                  }
+                })
+            .getText();
+    assertEquals(recentTask.get("nickname"), taskNeighbor);
+
+    String taskPoints =
+        wait.until(
+                new Function<WebDriver, WebElement>() {
+                  public WebElement apply(WebDriver driver) {
+                    return driver.findElement(By.xpath(taskCompletedXPath + "/td[4]"));
+                  }
+                })
+            .getText();
+    assertEquals(recentTask.get("points"), taskPoints);
+
+    verifyTaskDetails(taskCompletedXPath + "/td[1]");
+  }
+
+  @Test
+  /** Tests functionality of neighbor user verifying that helper did indeed complete a task */
+  public void _07_Userpage_AsLoggedUser_VerifyCompletedTask() {
+    // Logs out by using 'logout-href' element id;
+    logOut("logout-href");
+    loginUser(USER_EMAIL);
+    ifLaggingThenRefresh();
+
+    // Location of first task listed in the await verification table
+    String awaitVerifTaskXPath = "//tbody[@id='await-verif-body']/tr[1]";
+
+    String taskOverview =
+        wait.until(
+                new Function<WebDriver, WebElement>() {
+                  public WebElement apply(WebDriver driver) {
+                    return driver.findElement(By.xpath(awaitVerifTaskXPath + "/td[1]"));
+                  }
+                })
+            .getText();
+    assertEquals(recentTask.get("overview"), taskOverview);
+
+    String taskHelper =
+        wait.until(
+                new Function<WebDriver, WebElement>() {
+                  public WebElement apply(WebDriver driver) {
+                    return driver.findElement(By.xpath(awaitVerifTaskXPath + "/td[2]"));
+                  }
+                })
+            .getText();
+    assertEquals(recentTask.get("helper"), taskHelper);
+
+    String taskStatus =
+        wait.until(
+                new Function<WebDriver, WebElement>() {
+                  public WebElement apply(WebDriver driver) {
+                    return driver.findElement(By.xpath(awaitVerifTaskXPath + "/td[3]"));
+                  }
+                })
+            .getText();
+    assertEquals(recentTask.get("status"), taskStatus);
+
+    verifyTaskDetails(awaitVerifTaskXPath + "/td[1]");
+
+    // click button to verify a task has been completed
+    wait.until(
+            new Function<WebDriver, WebElement>() {
+              public WebElement apply(WebDriver driver) {
+                return driver.findElement(By.xpath(awaitVerifTaskXPath + "/td[4]/button"));
+              }
+            })
+        .click();
+
+    // Clicking on verify button triggers an alert confirmation window
+    // Driver will try to accept the alert to verify the task every second for a minute
+    for (int i = 0; i < 60; i++) {
+      try {
+        driver.switchTo().alert().accept();
+        break;
+      } catch (NoAlertPresentException e) {
+        driver.manage().timeouts().implicitlyWait(1, TimeUnit.SECONDS);
```

Comment:
I'm a little bit confused about all these `implicitlyWait` functions here. Are they really necessary? If they are, can we reduce the time we need to wait for?

Reply:
Unfortunately the webdriver is very unpredictable and without the waits it errors out more often. The examples I saw online typically have like 30seconds of wait for things - which I thought was too much, but 5-10 seconds doesn't seem like that much. 

In this particular piece of code it really isn't waiting longer than necessary, it's just polling constantly checking to see if the alert has shown up yet (or is accessible by the driver), waiting a second and then trying again. If it succeeds, it breaks, but at the most it takes 60seconds before erroring out. Because the alerts are considered a different "window" of sorts by the browser, this type of while loop helps the webdriver effectively find the alert. At least that's what I found from some suggestions online and it seemed to help as I was running into more errors before doing this.